src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+function floor(square: number): number {
+  return Math.floor(square * 100) / 100;
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: Shape,
+  color: string,
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  color: string;
+
+  constructor(
+    color:string, public a: number, public b: number, public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle sides must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Triangle sides arent correct');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
```

Comment:
```suggestion
  constructor(
    public color:string,
    public a: number,
    public b: number,
    public c: number,
  ) {
    if (a <= 0 || b <= 0 || c <= 0) {
      throw new Error('Triangle sides must be greater than 0');
    }

    if (a + b <= c || a + c <= b || b + c <= a) {
      throw new Error('Triangle sides arent correct');
    }
  }
```

Reply:
writing `public` before constructor param is a typescript feature,
it's already assigning value to this.{paramname}

(public a:string) equal to (a: string) { this.a = a }